api/src/main/java/org/apache/iceberg/transforms/ProjectionUtil.java

```diff
@@ -94,7 +94,56 @@ private ProjectionUtil() {}
     }
   }
 
-  static UnboundPredicate<Integer> truncateLongStrict(
+  static <T> UnboundPredicate<T> truncateIntegerStrict(
+      String name, BoundPredicate<Integer> pred, Transform<Integer, T> transform) {
```

Comment:
Why was it necessary to add these new methods? Did the original version not work?

Reply:
Yeah, it won't because it's assuming that the partition can be increased/decreased by 1. I was thinking of getting rid of the earlier methods and use these ones instead. The logic can then be shared, will just update the Timestamps and Dates Projection tests.